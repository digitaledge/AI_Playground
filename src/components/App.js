import React from 'react';
import PropTypes from 'prop-types';
import { Provider as ReduxProvider } from 'react-redux';

const ContextType = {
  insertCss: PropTypes.func.isRequired,
  fetch: PropTypes.func.isRequired,
  pathname: PropTypes.string,
  query: PropTypes.object,
  ...ReduxProvider.childContextTypes,
};

class App extends React.PureComponent {
  static propTypes = {
    context: PropTypes.shape(ContextType).isRequired,
    children: PropTypes.element.isRequired,
  };

  static childContextTypes = ContextType;

  getChildContext() {
    return this.props.context;
  }

  render() {
    return React.Children.only(this.props.children);
  }
}

export default App;
