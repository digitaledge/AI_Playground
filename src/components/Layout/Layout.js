/* @flow */
import React from 'react';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import normalizeCss from 'normalize.css';

import s from './Layout.css';
import Header from '../Header';

type Props = {
  children: any,
};

class Layout extends React.Component<Props> {
  render() {
    return (
      <div>
        <Header />
        {this.props.children}
      </div>
    );
  }
}

export default withStyles(normalizeCss, s)(Layout);
