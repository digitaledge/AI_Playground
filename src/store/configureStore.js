import thunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { name, version } from '../../package.json';
import rootReducer from '../reducers';
import createHelpers from './createHelpers';
import createLogger from './logger';

export default function configureStore(initialState, helpersConfig) {
  const helpers = createHelpers(helpersConfig);
  const middleware = [thunk.withExtraArgument(helpers)];

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    const composeEnhancers = composeWithDevTools({
      name: `${name}@${version}`,
    });

    enhancer = composeEnhancers(applyMiddleware(...middleware));
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  const store = createStore(rootReducer, initialState, enhancer);

  if (__DEV__ && module.hot) {
    module.hot.accept('../reducers', () =>
      // eslint-disable-next-line global-require
      store.replaceReducer(require('../reducers').default),
    );
  }

  return store;
}
