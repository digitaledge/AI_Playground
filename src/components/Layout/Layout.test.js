/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */
import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import App from '../App';
import Layout from './Layout';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};

describe('Layout', () => {
  test('Layout does render with children', () => {
    const store = mockStore(initialState);
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {}, fetch: () => {}, store }}>
          <Layout>
            <div className="child" />
          </Layout>
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
