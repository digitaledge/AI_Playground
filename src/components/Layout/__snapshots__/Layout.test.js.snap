// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Layout does render with children 1`] = `
<div>
  <div
    className="root"
  >
    <div
      className="container"
    >
      <div
        className="banner"
      >
        <h1
          className="bannerTitle"
        >
          AI Playground
        </h1>
        <p
          className="bannerDesc"
        >
          Node.js + OpenCV + React = fun!
        </p>
      </div>
    </div>
  </div>
  <div
    className="child"
  />
</div>
`;
